crates/edr_napi/src/miner.rs

```diff
@@ -46,6 +49,12 @@ pub async fn mine_block(
     let base_fee: Option<U256> =
         base_fee.map_or(Ok(None), |base_fee| BigInt::try_cast(base_fee).map(Some))?;
     let prevrandao: Option<B256> = prevrandao.map(TryCast::<B256>::try_cast).transpose()?;
+    if config.spec_id >= SpecId::MERGE && prevrandao.is_none() {
```

Comment:
I removed the check from `edr_evm`, because we need to be able to mine a pending block without the mix hash in `ProviderData`: https://github.com/NomicFoundation/hardhat/pull/4746/files/cac9a3d16f9fbc2208a0f2f5ebe6da4f051b8b04#diff-03b2ca6fd82ba7931a70b3b1ef672aa25b18c86bf643d21a930efa2691e701cfR1318

Not sure about the implications for napi

Reply:
After having a closer inspection, it seems fine. This implementation of `mine_block` also still contains the insertion of the block into the blockchain, so it's in mine with what's excepted.